### PR TITLE
feat: add a presenters' countdown view that can be used as an overlay

### PR DESCRIPTION
**Pull request by jstarpl**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

This is a new feature

* **What is the current behavior?** (You can also link to an open issue here)

Sofie has a single countdown screen that is suitable for full-screen display. It shows information about the current part, the countdown to the end of the current part, information about the next part, the name of the current rundownPlaylist and an over/under DIFF timer.

* **What is the new behavior (if this is a feature change)?**

A new countdown screen is added under `/countdowns/{:studioId}/overlay` that is suitable to be overlayed on top of an external video signal. It shows the countdown to the end of the current part, next part information (logo and piece name of the "onPresenterScreen" piece) and a wall clock. The view is responsive and can be displayed on any screen size and resolution.

obraz

* **Other information**:

This feature requires the blueprints to set up a useful CasparCG output with this view.

**Status**

- [x] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/client/styles/countdown/overlay.scss [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	top: 0;
+	bottom: 0;
+	left: 0;
+	right: 0;
+
+	height: 100vh;
+	width: 100vw;
+	// this ensures that no matter the screen size, the same amount of information can be displayed
+	// since the font-size is dependant on screen width
+	font-size: 5.6vw;
```
Comment (gundelsby) [inline meteor/client/styles/countdown/overlay.scss:18]: Nice 👍

File: meteor/client/ui/ClockView/OverlayScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+								for (const item of segment.items) {
+									if (item.instance._id === playlist.nextPartInstanceId) {
+										nextSegment = segment
+										nextPart = item
+									}
+								}
+							}
+						}
+
+						// const overUnderClock = playlist.expectedDuration
```
Comment (gundelsby) [inline meteor/client/ui/ClockView/OverlayScreen.tsx:150]: Dead code. Should either be deleted, or commented to tell why it should be saved for later

File: meteor/client/ui/ClockView/OverlayScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+							<div className="clocks-overlay">
+								<div className="clocks-half clocks-bottom">
+									<div className="clocks-current-segment-countdown clocks-segment-countdown">
+										{currentPartDuration !== null ? (
+											<Timediff time={currentPartDuration} />
+										) : (
+											<span className="clock-segment-countdown-next">{t('Next')}</span>
+										)}
+									</div>
+									{/* {currentPart && currentPart.instance.part.autoNext ? (
```
Comment (gundelsby) [inline meteor/client/ui/ClockView/OverlayScreen.tsx:167]: Dead code

File: meteor/client/ui/i18n.ts [hunk 4]
```diff
@@ -45,6 +45,7 @@ const i18nInstancePromise = i18n
 			} else {
 				i18nTranslator = t
 				moment.locale(i18n.language)
+				document.documentElement.lang = i18n.language
```
Comment (gundelsby) [inline meteor/client/ui/i18n.ts:48]: Good shout, will keep that in mind for when I do the translations update later 👍

**Review by gundelsby — APPROVED** [review]
LGTM :)

I'd prefer the two instances of dead code to be deleted or commented as to why they should be kept

File: meteor/client/ui/ClockView/OverlayScreen.tsx (hunk 2 above)
Comment (jstarpl) [reply meteor/client/ui/ClockView/OverlayScreen.tsx:152]: I've added some comments to tell why I left these parts in for future reference.

**Pull request merged**